lib/decorators/inject.ts

```diff
@@ -75,5 +76,27 @@ export function Inject( ...injects: any[] ){
 		else {
 			bundleStore.set('$inject', dependencies, t);
 		}
+
+		componentHooks.beforeCtrlInvoke(injectParentComponents);
 	}
 }
+
```

Comment:
Outside to this top level scope so the hook is only added once. The hook function will be called once per every component.